src/main/scala/za/co/absa/commons/buildinfo/BuildInfo.scala

```diff
@@ -29,33 +29,22 @@ import za.co.absa.commons.lang.OptionImplicits._
  * file to your classpath (without 'template' suffix) and enable Maven resource filtering.
  *
  * If BuildInfo defaults don't suit your project you can create your own instance
- * and parametrize it in one of the following ways:
+ * and parametrize it:
  * {{{
- *   object MyBuildInfo extend BuildInfo(...)
- * }}}
- * or
- * {{{
- *   val myBuildInfo = BuildInfo(...)
+ *   object MyBuildInfo extend BuildInfoTemplate(...)
  * }}}
+ *
  */
-object BuildInfo extends BuildInfo(
-  resourcePrefix = BuildInfoConst.DefaultResourcePrefix,
-  propMapping = PropMapping.Default) {
-
-  def apply(
-    resourcePrefix: String = BuildInfoConst.DefaultResourcePrefix,
-    propMapping: PropMapping = PropMapping.Default
-  ): BuildInfo = new BuildInfo(resourcePrefix, propMapping) {}
-}
+object BuildInfo extends BuildInfoTemplate
 
-object BuildInfoConst {
-  // it has to be separated from the `BuildInfo` object due to a bug: https://github.com/scala/bug/issues/5000
+object BuildInfoTemplate {
```

Comment:
Why this `object`? 
And why a `BuildInfoTemplate` class at all. Cannot you just have a `BuildInfo` class (and `BuildInfo` object extending it)?

Reply:
I guess these two fields could be directly in constructor of BuildInfoTemplate as default values.

I don't want to have the same object extend the class and be companion. Since it then can cause the companion object be created even though you don't want to use it.